connect/mirror/src/test/java/org/apache/kafka/connect/mirror/integration/MirrorConnectorsIntegrationTest.java

```diff
@@ -396,27 +330,67 @@ public void testOneWayReplicationWithAutoOffsetSync() throws InterruptedExceptio
             "group.id", "consumer-group-1"), "primary.test-topic-1", "primary.test-topic-2");
 
-        waitForConsumerGroupOffsetSync(consumer, Arrays.asList("primary.test-topic-1", "primary.test-topic-2"), "consumer-group-1");
+        waitForConsumerGroupOffsetSync(backup, consumer, Arrays.asList("primary.test-topic-1", "primary.test-topic-2"), 
+            "consumer-group-1", NUM_RECORDS_PRODUCED);
 
         records = consumer.poll(Duration.ofMillis(500));
         // similar reasoning as above, no more records to consume by the same consumer group at backup cluster
         assertEquals("consumer record size is not zero", 0, records.count());
         consumer.close();
-
     }
+    
+    @Test
+    public void testWithBrokerRestart() throws InterruptedException {
+        // test with a higher number of records
+        int numRecords = NUM_RECORDS_PRODUCED * 100;
+        
+        produceRecords(Arrays.asList(primary), Arrays.asList("test-topic-1"), numRecords);
 
-    private void deleteAllTopics(EmbeddedKafkaCluster cluster) {
-        Admin client = cluster.createAdminClient();
-        try {
-            client.deleteTopics(client.listTopics().names().get());
-        } catch (Throwable e) {
-        }
+        // one way replication from primary to backup
+        mm2Props.put("backup->primary.enabled", "false");
+        mm2Config = new MirrorMakerConfig(mm2Props);
+       
+        waitUntilMirrorMakerIsRunning(backup, SOURCE_CONNECTOR, mm2Config, "primary", "backup");
+        
+        // have to sleep a little for MM to be ready for the following the kafka broker restart
+        Thread.sleep(TimeUnit.SECONDS.toMillis(1));
+
+        // restart kafka broker at backup cluster
+        restartKafkaBroker(backup);
+        
+        Consumer<byte[], byte[]> consumer = backup.kafka().createConsumerAndSubscribeTo(
+                Collections.singletonMap("group.id", "consumer-group-1"), "primary.test-topic-1");
+        // verify the consumption equals to produce
+        waitForConsumingAllRecords(consumer, numRecords);
+        consumer.commitAsync();
+        
+        // produce another set of records
+        produceRecords(Arrays.asList(primary), Arrays.asList("test-topic-1"), numRecords);
+        // restart kafka broker at primary cluster
+        restartKafkaBroker(primary);
+        // verify the consumption equals to produce
+        waitForConsumingAllRecords(consumer, numRecords);
+        
+        consumer.close();
+    }
+    
+    void createTopics() {
+        // to verify topic config will be sync-ed across clusters
+        Map<String, String> topicConfig = new HashMap<>();
+        topicConfig.put(TopicConfig.CLEANUP_POLICY_CONFIG, TopicConfig.CLEANUP_POLICY_COMPACT);
+        // create these topics before starting the connectors so we don't need to wait for discovery
+        primary.kafka().createTopic("test-topic-1", NUM_PARTITIONS, 1, topicConfig);
```

Comment:
when creating `test-topic-1` topic on primary cluster, add a topic config. Later on, we will check if the config is synced from primary to backup cluster.